internal/settings.go

```diff
@@ -187,3 +188,32 @@ func (ds *DialSettings) GetUniverseDomain() string {
 func (ds *DialSettings) IsUniverseDomainGDU() bool {
 	return ds.GetUniverseDomain() == ds.GetDefaultUniverseDomain()
 }
+
+// GetUniverseDomain returns the default service domain for a given Cloud
+// universe, from google.Credentials, for comparison with the value returned by
+// (*DialSettings).GetUniverseDomain. This wrapper function should be removed
+// to close [TODO(chrisdsmith): issue link here]. See details below.
+func GetUniverseDomain(creds *google.Credentials) (string, error) {
+	timer := time.NewTimer(time.Second)
+	defer timer.Stop()
+	var credsUniverseDomain string
+	var err error
+	go func() {
+		credsUniverseDomain, err = creds.GetUniverseDomain()
+	}()
+	<-timer.C
+	if err != nil {
```

Comment:
This code is racy. You are reading values that are updated in another goroutine without synchronization. I would recommend using channels and a select statement.

Reply:
Updated, PTAL.